modules/log/log.go

```diff
@@ -297,6 +302,7 @@ func (l *Logger) Close() {
 			break
 		}
 	}
+	l.lock.Unlock()
```

Comment:
I guess I have bug here as the lock encloses synchronization points (is this the correct terminology?).